mk_mvc.mak

```diff
@@ -9,9 +9,14 @@
 
 include source.mak
 
+OBJEXT = obj
+
+# to be replaced by gnulib/regex.{ch}
+REGEX_HEADS = gnu_regex/regex.h
+REGEX_SRCS = gnu_regex/regex.c
+
 REGEX_DEFINES = -DHAVE_REGCOMP -D__USE_GNU -DHAVE_STDBOOL_H -Dstrcasecmp=stricmp
 
-OBJEXT = obj
```

Comment:
Do we need this change?

Reply:
No. These `OBJEXT` related changes are not mandatory.

I think a Makefile variable should be defined before it is referred for readability, even if it is a _recursively expanded variable_ (cf. https://www.gnu.org/software/make/manual/make.html#Flavors).

So I moved the line before the following line I added;

```
GNULIB_OBJS = $(GNULIB_SRCS:.c=.$(OBJEXT))
```

(I did this on both mk_mingw.mak and mk_mvc.mak.  But `nmake` does not support `$(GNULIB_SRCS:.c=.$(OBJEXT))`. So there is no need for mk_mvc.mak.)

But I found `$(OBJEXT)` is also referred in `include source.mak`. I had to move the line `OBJEXT = obj` before the line `include source.mak`.

Which do you prefer?

1. do not change. (revert the change)
2. Move the line `OBJEXT = obj` before the line `include source.mak`.
3. Move the line `OBJEXT = obj` before the line `include source.mak` in a separate commit or PR.

Reply:
> No. These OBJEXT related changes are not mandatory.
> 
> I think a Makefile variable should be defined before it is referred for readability, even if it is a recursively expanded variable (cf. https://www.gnu.org/software/make/manual/make.html#Flavors).
> 

The change for improving readability is really appreciated. However, please do it in a separate commit.
So the change I have to review becomes smaller.
See  also
https://www.kernel.org/doc/html/v4.17/process/submitting-patches.html#separate-your-changes .

> 1. ...
> 2. ...
> 3. ...

I have no preference. Please, do as you think better. What I want is only the separation.

Reply:
I move the changes for OBJEXT in a separate commit, "build: move OBJEXT above `include source.mak'".